src/lib.rs

```diff
@@ -273,15 +295,22 @@ fn generate_function(closure: &ExprClosure, expressions: &FormattedAttributes, r
     let code = if result {
         quote! {
             fn temp() {
-                (#closure)()
-                    .map(|result| { #ok_expr; result })
-                    .map_err(|err| { #err_expr; err })
+                let begin = std::time::SystemTime::now();
+                let result = (#closure)();
+                let duration_0 = std::time::Duration::new(0, 0);
+                let ts = std::time::SystemTime::now().duration_since(begin).unwrap_or(duration_0);
+                let result = result.map(|result| { #ok_expr; result })
+                    .map_err(|err| { #err_expr; err });
+                result
```

Comment:
You should use `Instant` and `elapsed() -> Duration`

Reply:
And you just added the timing unconditionally. if you pass false it will still time the function and not print.
That's a huge overhead.

Reply:
If pass false to it, the "ts" variable will not be used. Whether it will be optimized out by the compiler in release build? I don't know much about compilers.

Reply:
Thanks you for your advice.
I have modified the pull request.